### DISCUSSION TITLE
Inference flow from input to loss.

### DISCUSSION THREAD

**Discussion by 170928** [Arthur151/ROMP#342] [Q&A]
Hi @Arthur151 , I am trying to understand BEV in detail and make various attempts.
So, I succeeded in understanding the great ideas and implementation of Mixed_dataset.

I also understood the loss function, but I learned that 'outputs' delivered for training in train.py are not 'outputs' of BEV Module.
I thought the values ​​of outputs were 'params_pred', 'cam_czyx', 'center_map', .... but I was able to guess that it is actually the output of result_parser . 

```python
def train_step(self, meta_data):
     ...
     # NOTE I think outputs are BEV module's output, but it is not...
     # 'params_pred', 'cam_czyx', 'center_map', 'center_map_3d' .....
     if not self.model_return_loss:
          outputs.update(self._calc_loss(outputs))
     loss, outputs = self.merge_losses(outputs(outputs, self.train_cfg['new_training'])
     ...
```

I have a question here :(
Can you tell what functions the output of BEV goes through in the process of being passed to the code below?

(ex) 
```python
outputs = BEV(images) 
outputs = ResultParser.matching_forward(outputs)
self.merge_losses(outputs)
```

Sorry for trivial questions T _ T 
I really like BEV and want to use this! Thus, I have a lot of questions haha...

**Comment by Arthur151** [comment]
Hi, MJ Shin, @170928 
Great question!
The data flow of `output` from BEV model to the loss function is manipulated by the base model
https://github.com/Arthur151/ROMP/blob/b108da9e9fe96282d8235a1051a7030d335d9869/romp/lib/models/base.py#L33
I think this function will clarify your doubts. 
Please don't hestitate to let me know your question. 
Best,
Yu
